L1Trigger/Phase2L1ParticleFlow/plugins/L1CTJetFileWriter.cc

```diff
@@ -0,0 +1,109 @@
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Common/interface/View.h"
+
+#include "L1Trigger/DemonstratorTools/interface/BoardDataWriter.h"
+#include "L1Trigger/DemonstratorTools/interface/utilities.h"
+#include "DataFormats/L1TParticleFlow/interface/PFJet.h"
+#include "L1Trigger/Phase2L1ParticleFlow/src/newfirmware/dataformats/gt_datatypes.h"
+
+//
+// class declaration
+//
+
+class L1CTJetFileWriter : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+public:
+  explicit L1CTJetFileWriter(const edm::ParameterSet&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  // ----------constants, enums and typedefs ---------
+  unsigned nJets;
+  size_t kFramesPerBX; 
+  size_t kCTL2BoardTMUX; 
+  size_t kGapLengthOutput; 
+  size_t kMaxLinesPerFile; 
+  std::map<l1t::demo::LinkId, std::pair<l1t::demo::ChannelSpec, std::vector<size_t>>> kChannelSpecsOutputToGT; 
+
+  // ----------member functions ----------------------
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+  std::array<std::vector<ap_uint<64>>, 1> encodeJets(const edm::View<l1t::PFJet>& jets);
+
+  edm::EDGetTokenT<edm::View<l1t::PFJet>> jetsToken_;
+  l1t::demo::BoardDataWriter fileWriterOutputToGT_;
+};
+
+L1CTJetFileWriter::L1CTJetFileWriter(const edm::ParameterSet& iConfig) :
+      nJets(iConfig.getParameter<unsigned>("nJets")),
+      kFramesPerBX(iConfig.getParameter<unsigned>("nFramesPerBX")),
+      kCTL2BoardTMUX(iConfig.getParameter<unsigned>("TMUX")),
+      kGapLengthOutput(kFramesPerBX - 2 * nJets),
```

Comment:
@cerminar pointed out this should instead be be `kCTL2BoardTMUX * kFramesPerBX - 2 * nJets`